mllib/src/main/scala/org/apache/spark/ml/tree/Node.scala

```diff
@@ -287,6 +292,41 @@ private[tree] class LearningNode(
     }
   }
 
+  /**
+   * Method testing whether a node is a leaf.
+   * @return true iff a node is a leaf.
```

Comment:
You don't need to doc a private method, but you can. But for my taste this is redundant. Just keep the `@return` doc

Reply:
I have to admit that I have overdone it a bit with documentation here, I have tried to reduce it all over the PR.